File: Books/A_Comprehensive_Approach_to_ML_and_AI/python/sigmoid_test.py
import numpy as np
import matplotlib.pyplot as plt

# Plot the standard sigmoid function.
x = np.arange(-8, 8, 0.1)
f = 1 / (1 + np.exp(-x))
plt.figure()
plt.plot(x, f, label='sigmoid(x)')
plt.xlabel("x")
plt.ylabel("f(x)")
plt.title("Sigmoid Function")
plt.legend()
plt.show()

# Adjusting the weights.
weights = [0.5, 1.0, 2.0]
plt.figure()
for w in weights:
    f_w = 1 / (1 + np.exp(-w * x))
    plt.plot(x, f_w, label=f"w = {w}")
plt.xlabel("x")
plt.ylabel("h_w(x)")
plt.title("Effect of Weight on Sigmoid")
plt.legend()
plt.show()

# Effect of bias.
w = 5.0
biases = [-8.0, 0.0, 8.0]
plt.figure()
for b in biases:
    f_wb = 1 / (1 + np.exp(-(w * x + b)))
    plt.plot(x, f_wb, label=f"b = {b}")
plt.xlabel("x")
plt.ylabel("h_wb(x)")
plt.title("Effect of Bias on Sigmoid")
plt.legend()
plt.show()

File: Books/A_Comprehensive_Approach_to_ML_and_AI/matlab/sigmoid_test.m
% Start of script
%-------------------------------------------------------------------------%
close all;  clear all; clc; 

% Simulate
%-------------------------------------------------------------------------%   
% neuron
x = (-8:0.1:8);
f = 1 ./ (1 + exp(-x));
figure;
plot(x, f)
xlabel('x')
ylabel('f(x)')
% savefig('sigmoid.png')

% adjusting the weights
w = [0.5,1.0,2.0];
l1 = 'w = 0.5';
l2 = 'w = 1.0';
l3 = 'w = 2.0';
figure;
for i=1:3
	f = 1 ./ (1 + exp(-x.*w(i)));
	plot(x, f);
    hold on
end
xlabel('x')
ylabel('h_w(x)')
legend(l1,l2,l3)
% savefig('Weight-adjustment-example.png')

% Effect of Bias
w = 5.0;
b = [-8.0,0.0,8.0];
l1 = 'b = -8.0';
l2 = 'b = 0.0';
l3 = 'b = 8.0';
figure;
for i=1:3
	f = 1 ./ (1 + exp(-(x*w+b(i))));
	plot(x, f);
    hold on
end
xlabel('x')
ylabel('h_wb(x)')
legend(l1,l2,l3)
% savefig('Bias-adjustment-example.png')